viewer/viewer-awt/src/main/java/org/icepdf/ri/common/SwingController.java

```diff
@@ -376,6 +376,13 @@ public WindowManagementCallback getWindowManagementCallback() {
      */
     public void setPropertiesManager(ViewerPropertiesManager propertiesManager) {
         this.propertiesManager = propertiesManager;
+        //Migrate old boolean property to new int property
```

Comment:
Ugh, never really had to think about migrating properties before.  How did you decide on PROPERTY_ANNOTATION_INK_SELECTION_ENABLED as the property to set off of?  What I might suggest as an alternative is the changing the properties keys.  The next release is going to be major so we can break anything we can get our hands on api wise.  Perhaps change the keys ICEpdfDEfault.properties  and similarly in ViewerPropertiesManager to something like this:

application.annotation.highlight.selection.enabled -> application.annotation.highlight.selection.type

And update the AnnotationHandlers accordingly.   I'll update the release notes  accordingly.

Reply:
Well, if I remember well I just picked one of them at random, assuming that most people would have the same value for every option.        
That'd be cleaner to do that indeed, I think it's a good idea. But I'm wondering if it makes sense to have a property for each annotation tool or if one for all of them would be sufficient?